src/main/webapp/app/controllers/discoveryContextController.js

```diff
@@ -124,12 +119,20 @@ sage.controller('DiscoveryContextController', function ($controller, $scope, $ro
       });
     };
 
-    $scope.searchProcessKeyPress = function(event) {
-      if (event.keyCode === 13 && $scope.currentSearchField) {
+    $scope.setCurrentSearchField = function(searchField) {
+      $scope.currentSearchField = searchField;
+    };
+
+    $scope.search = function() {
         $scope.discoveryContext.setSearchField($scope.currentSearchField.key, $scope.currentSearchValue, $scope.findSearchFieldLabel($scope.currentSearchField.key));
```

Comment:
Is it possible for `$scope.currentSearchField` to be undefined here? I assume you can press search without selecting a search field.

Reply:
Good catch. New commit initializes currentSearchField and resets it when the home button is clicked.